jupyter_notebook_parser/container.py

```diff
@@ -1,12 +1,13 @@
 import re
-from typing import Tuple, List, Dict
-
-ipython_line_magic = re.compile(r'^(%\S+)($|\s.*)')  # https://regex101.com/r/LTQpNc/1
-ipython_cell_magic = re.compile(r'^(%%\S+)($|\s.*)')
 
 
 class SourceCodeContainer:
-    def __init__(self, raw_source: str) -> None:
+    __ipython_line_magic = re.compile(r"^(%\S+)($|\s.*)")  # https://regex101.com/r/LTQpNc/1
+    __ipython_cell_magic = re.compile(r"^(%%\S+)($|\s.*)")
+
+    __slots__ = ("raw_source", "lines_without_magic", "magics", "source_without_magic")
+
+    def __init__(self, raw_source):
```

Comment:
Why did you delete type hints?

Reply:
Type hints moved to .pyi files which are designed for this

Reply:
Leaving type hints in place is good for readability, so I will not accept PRs that remove type hints from .py files.